core/src/main/java/org/apache/commons/vfs2/impl/DefaultFileSystemManager.java

```diff
@@ -715,10 +715,19 @@ public FileObject resolveFile(final FileObject baseFile, final String uri,
 					}
 					String proxyUser = queryParam.get(SftpConstants.PROXY_USERNAME);
 					String proxyPassword = queryParam.get(SftpConstants.PROXY_PASSWORD);
-
-					((SftpFileSystemConfigBuilder) (provider.getConfigBuilder())).setProxyType(fileSystemOptions,
-					                                                                           SftpFileSystemConfigBuilder.PROXY_HTTP);
-					((SftpFileSystemConfigBuilder) (provider.getConfigBuilder())).setProxyHost(fileSystemOptions,
+                    String proxyType = queryParam.get(SftpConstants.PROXY_TYPE);
+
+                    if (SftpConstants.SOCKS.equals(proxyType)) {
+                        ((SftpFileSystemConfigBuilder) (provider.getConfigBuilder()))
+                                .setProxyType(fileSystemOptions, SftpFileSystemConfigBuilder.PROXY_SOCKS5);
+                    } else {
+                        if (proxyType != null && !"".equals(proxyType) && !SftpConstants.HTTP.equals(proxyType)) {
```

Comment:
We should use proper parenthesis for clarity of the logic. 
Can we use String.isEmpty() instead ?